sdk/storage/Azure.Storage.DataMovement.Files.Shares/tests/DataMovementShareTestBase.cs

```diff
@@ -0,0 +1,86 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System.Threading.Tasks;
+using System.Collections.Generic;
+using Azure.Core.TestFramework;
+using Azure.Storage.Files.Shares;
+using Azure.Storage.Files.Shares.Tests;
+using Azure.Storage.Test.Shared;
+using SharesClientBuilder = Azure.Storage.Test.Shared.ClientBuilder<
+    Azure.Storage.Files.Shares.ShareServiceClient,
+    Azure.Storage.Files.Shares.ShareClientOptions>;
+using System.IO;
+
+namespace Azure.Storage.DataMovement.Files.Shares.Tests
+{
+    [ShareClientTestFixture]
+    public abstract class DataMovementShareTestBase : StorageTestBase<StorageTestEnvironment>
```

Comment:
Using a base test class per-package is a thing that's caused problems when we want to create tests that can work with multiple packages (something it seems data movement will absolutely want to do). Since we're writing from scratch here, we should avoid this altogether.

A lot of packages have the client builder and tenant credentials sort of retrofitted onto the class as a step of centralizing some of that logic for tests that already can break free of this test base paradigm, but we should really be using them *instead of* a package-specific base test class. We can't escape the need for Azure.Core's base test class as that's where all the recording logic is, but we can avoid locking ourselves into that paradigm here.

Reply:
done, moved it to specific base test class. Can probably move blobs over to this in the future as well.